actors/builtin/miner/testing.go

```diff
@@ -514,42 +523,50 @@ func CheckExpirationQueue(expQ ExpirationQueue, liveSectors map[abi.SectorNumber
 			} else {
 				acc.Addf("on-time expiration sector %d isn't live", n)
 			}
-
 			return nil
-		}); err != nil {
-			return err
-		}
+		})
+		acc.RequireNoError(err, "error iterating early sectors")
 
 		// Validate power and pledge.
+		var activeSectors, faultySectors map[abi.SectorNumber]*SectorOnChainInfo
+		var missing []abi.SectorNumber
+
 		all, err := bitfield.MergeBitFields(exp.OnTimeSectors, exp.EarlySectors)
 		if err != nil {
-			return err
-		}
-		allActive, err := bitfield.SubtractBitField(all, partitionFaults)
-		if err != nil {
-			return err
-		}
-		allFaulty, err := bitfield.IntersectBitField(all, partitionFaults)
-		if err != nil {
-			return err
-		}
-		activeSectors, missing, err := selectSectorsMap(liveSectors, allActive)
-		if err != nil {
-			return err
-		} else if len(missing) > 0 {
-			acc.Addf("active sectors missing from live: %v", missing)
-		}
-		faultySectors, missing, err := selectSectorsMap(liveSectors, allFaulty)
-		if err != nil {
-			return err
-		} else if len(missing) > 0 {
-			acc.Addf("faulty sectors missing from live: %v", missing)
+			acc.Addf("error merging all on-time and early bitfields: %v", err)
+		} else {
+			if allActive, err := bitfield.SubtractBitField(all, partitionFaults); err != nil {
+				acc.Addf("error computing active sectors: %v", err)
+			} else {
+				activeSectors, missing, err = selectSectorsMap(liveSectors, allActive)
+				if err != nil {
+					acc.Addf("error selecting active sectors: %v", err)
```

Comment:
Elsewhere you've used the pattern of setting a datastructure as empty when there is an error loading or constructing it. If you did that for `all`, `allActive`, `allFaulty`, etc. you could avoid this nesting and I believe it all works so no redundant errors are produced.

Reply:
I went this way because those structures are bitfields and testing `Empty()` produces yet another error to handle. Maybe using `==` with the zero value would work? I'm a bit hesitant as it's otherwise a broken value.